tests/std/tests/GH_005204_regex_collating_ranges/test.cpp

```diff
@@ -562,9 +562,33 @@ void test_gh_994() {
 #endif // !defined(SKIP_COLLATE_TESTS)
 }
 
+void test_gh_5437_ECMAScript_or_collate(syntax_option_type ECMAScript_or_collate) {
+    {
+        test_wregex char_range(&g_regexTester, L"^[\u0001-\u0200]$", ECMAScript_or_collate);
+        for (wchar_t ch = L'\u0001'; ch <= L'\u0200'; ++ch) {
+            char_range.should_search_match(wstring(1, ch), wstring(1, ch));
+        }

```

Comment:
I would feel better if we had negative tests before and after the range. (Same principle as when testing things like type traits - if you only check for true answers, you'll miss bugs that cause the answer to always be true.)